src/find_libpython/__init__.py

```diff
@@ -277,6 +301,12 @@ def candidate_paths(suffix=_SHLIB_SUFFIX):
     for basename in lib_basenames:
         yield _find_library(_library_name(basename))
 
+    if _is_posix:
+        try:
+            yield from _get_proc_library()
+        except:
+            _logger.debug(f"Unable to check /proc filesystem for libpython")
+            
```

Comment:
I don't understand. Why are we doing it twice?

Reply:
I thought I should add the name to the list of candidate_names as well for completness. Not sure if it's a good idea though. Having the code in candidate_paths makes sense to me.

Reply:
The distinction is pointless IMO (I inherited that) and both `--candidate-names` and `--candidate-paths` should be removed at some point for a more holistic search algorithm. I'd remove the bit from the `candidate_names` function, as that is really only for guessing the names of the libpython object for dynamic lookup. Your function returns a path which is infinitely more useful.